docs/designs/ps_design.md

```diff
@@ -1,377 +1,94 @@
 
-Each worker has a local copy of the model variables. After the master relaunches a PS node, the PS node can recover model variables from workers. For embedding vectors, PS must create replicas to support fault tolerance. For each PS node *PS(i)*, it will store *M* replicas in the following *M* PS nodes from *PS(i+1 % N)* to *PS(i+M % N)*. The relaunched PS node can recover embedding vectors from one of its replicas. If there are more than *M* continuously-indexed PS nodes failing, at least one PS node fails with all of its replicas. The ElasticDL job has to recover from a recent checkpoint.
+Thus, a distributed PS can support the scalability of model size. it also supports the scalability of training data size by providing enough bandwidth and gradient processing capacity for large number of workers.
 
-## PS
+## PS Fault Tolerance
+ElasticDL is a fault-tolerable deep learning system. A distributed PS consists of multiple PS pods. Each PS pod stores a partition of model parameters. Workers need to get model parameters from all PS pods in the forward-pass of each iteration. A failed PS pod will interrupt the training. We can relaunch any failed PS pod and recover the partition of model parameters on it to support PS fault tolerance.
 
-### Compoments
+In ElasticDL, a master is responsible for creating PS pods and worker pods using Kubernetes APIs. The master launches PS pods with high priority and launches worker pods with low priority. The master also monitor PS pods and worker pods status. In case a PS pod fails, the master will relaunch it using Kubernetes APIs. Since PS pods have a higher priority than worker pods, if there are still some running the worker pods, the relaunch will succeed by using either idle or preempted Kubernetes resources. If no worker pods left, ElasticDL has to wait for Kubernetes resources to continue the training.
 
-![pserver](../images/pserver.png)
+After the relaunch of a PS pod, the PS pod needs to recover its partition of model parameters. For a worker, a minibatch of training data in a forward-pass contains some embedding vectors, which is a subset of embedding tables (the parameters of the corresponding embedding layers). The worker pulls all non-embedding parameters and a subset of embedding tables from PS pods in the training. The PS pod can recover non-embedding parameters from workers but not embedding tables.
 
-PS contains two main compoments:
+In order to recover the embedding table partition in the relaunched PS pod, the distributed PS needs to have embedding table replicas. A PS pod can store its embedding table partition replicas in other PS pods. For example, assume there are *N* PS pods from *PS<sub>0</sub>* to *PS<sub>N-1</sub>*, *PS<sub>i</sub>* can stores its replica in *PS<sub>(i + 1) % N</sub>*. The relaunched PS pod  *PS<sub>i</sub>* can recover its embedding table partition from its replica in  *PS<sub>(i + 1) % N</sub>*.
```

Comment:
It might not be a good idea to use HTML tag <sub> in Markdown files. Some renderers do not recognize such tags. It also makes it less comprehensive to readers. I would usually Google "subscript x" and paste the Unicode points in the Markdown files.

Reply:
Mac cannot display some Unicode of alphabet subscripts correctly, such as 'm'.
The `<sub>` tag works fine on github and on Mac.